hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -629,7 +629,6 @@ public boolean delete(Data key) {
 
         final Record record = getRecordOrNull(key, now, false);
         if (record == null) {
-            removeIndex(record);
```

Comment:
this must have been some left-over not covered by test. 
The `removeIndex()` really does not make sense when the record is not stored locally. This change makes it aligned with the `remove()` method.